src/main/java/com/conveyal/r5/streets/StreetLayer.java

```diff
@@ -1289,9 +1290,9 @@ public int createAndLinkVertex (double lat, double lon) {
             return -1; // Unlinked
         }
 
-        // TODO give link edges a length.
+        int length_mm = (int) (GeometryUtils.distance(lat,lon, vertexStore.getCursor(streetVertex).getLat(), vertexStore.getCursor(streetVertex).getLon())*1000);
```

Comment:
This is creating two vertex cursor instances - again this is an arguably minor optimization, but it's generally better to avoid creating excess objects in heavily used functions.

My take is that it's worth constructing the Vertex cursor because that's our project's standard way of accessing the details of vertices, but we don't want to make extra ones. Who knows, the JVM might be able to optimize this away, but nonetheless I think the meaning is clearer if you make one such object representing the vertex, then read off the lat and lon from the same object.